### PR TITLE
#86a0e97pz update diatheke python examples and add ASR stream with partial

### PR DESCRIPTION
**Pull request by ppcssn**

### CODE REVIEW

File: diatheke/audio_client.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 #
-# Copyright(2021) Cobalt Speech and Language Inc.
+# Copyright (2020 -- present) Cobalt Speech and Language, Inc.
```
Comment (mkylemartin) [inline diatheke/audio_client.py:4]: can we put 2023 present? maybe to give the impression that this script isn't 3 years old?

File: diatheke/streams.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        if not self._has_result():
+            self._data_queue.put(None)
+
+        # Wait for the streaming thread to finish and return the result.
+        self._stream_thread.join()
+        return self._result
+
+
+def read_ASR_audio(stream, reader, buffSize):
+    """Deprecated. Use client.read_asr_audio() instead."""
```
Comment (mkylemartin) [inline diatheke/streams.py:135]: why are we including these if they're deprecated?

File: diatheke/streams.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    else:
+        # Otherwise we assume we have a byte writer
+        is_text = False
+
+    for data in stream:
+        if is_text:
+            # Convert the text to a string before writing.
+            writer.write(str(data.audio))
+        else:
+            writer.write(data.audio)
```
Comment (mkylemartin) [inline diatheke/streams.py:171]: same comment as above (why keep if deprecated)

File: diatheke/streams.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+
+def read_ASR_audio(stream, reader, buffSize):
+    """Deprecated. Use client.read_asr_audio() instead."""
+    # Check if we have a text reader
+    if isinstance(reader, io.TextIOBase):
+        is_text = True
+    else:
+        # Otherwise we assume we have a byte reader
+        is_text = False
```
Comment (mkylemartin) [inline diatheke/streams.py:141]:
```suggestion
    # Assume we have a byte reader
    is_text = False
    # assign text reader if it's that
    if isinstance(reader, io.TextIOBase):
        is_text = True
```

Whatever one is the most expected/common should be set and then the if statement can check on the alternative case. 

Mostly a style/readability suggestion/

File: diatheke/streams.py [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+        # streaming method.
+        def wait_for_audio():
+            while not is_done():
+                # Wait for data from the queue, which will block until data
+                # is available.
+                request = input.get()
+                if request is None:
+                    # This is our sentinel to indicate that the generator
+                    # should finish, which will end the stream.
+                    return
```
Comment (mkylemartin) [inline diatheke/streams.py:218]:
```suggestion
                    return None
```
[`return` vs `return None` should be the same but just for sake of readability, i prefer `return None`](https://stackoverflow.com/a/15300733)

**Review by mkylemartin — CHANGES_REQUESTED** [review]
looks good to me; just a few minor changes / requests.

File: diatheke/streams.py (hunk 2 above)
Comment (ppcssn) [reply diatheke/streams.py:135]: Good question. I don't know the answer to this either, but Original file is [here](https://github.com/cobaltspeech/sdk-diatheke/blob/master/grpc/py-diatheke/diatheke/streams.py).

File: diatheke/streams.py (hunk 3 above)
Comment (ppcssn) [reply diatheke/streams.py:171]: probably relate to the reason as `read_ASR_audio`

File: diatheke/streams.py (hunk 5 above)
Comment (ppcssn) [reply diatheke/streams.py:218]: Good point. It aligns with [PEP8](https://peps.python.org/pep-0008/#programming-recommendations),

> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable):

and we also do it in other typed programming languages, so I guess this make sense.

File: diatheke/audio_client.py [hunk 6]
```diff
@@ -50,30 +50,37 @@ def wait_for_input(client, session, input_action):
...
-    
+
     if input_action.requires_wake_word:
         # This action requires the wake-word to be spoken before
         # the user input will be accepted. Use a wake-word detector
         # and wait for it to trigger.
         pass
 
+    def result_handler(result):
+        if len(result.partial_result.alternatives) != 0:
```
Comment (happyalu) [inline diatheke/audio_client.py:61]: will this throw an exception if partial_result is None ?

**Review by happyalu — APPROVED** [review]

File: diatheke/audio_client.py (hunk 6 above)
Comment (ppcssn) [reply diatheke/audio_client.py:61]: no, it won't throw anything, that why I need to check the alternatives length

**Pull request merged**